common/lib/common-utils/src/bufferBrowser.ts

```diff
@@ -92,10 +104,11 @@ export class IsoBuffer extends Uint8Array {
             return IsoBuffer.fromString(value, encodingOrOffset as string | undefined);
         // Capture any typed arrays, including Uint8Array (and thus - IsoBuffer!)
         } else if (value !== null && typeof value === "object" && isArrayBuffer(value.buffer)) {
-            // Support currently for full array, no view ports! (though it can be added in future)
-            assert(value.byteOffset === 0, 0x000 /* "nonzero isobuffer byte offset" */);
-            assert(value.byteLength === value.buffer.byteLength, 0x001 /* "unexpected isobuffer byte length" */);
-            return IsoBuffer.fromArrayBuffer(value.buffer, encodingOrOffset as number | undefined, length);
+            const offset = encodingOrOffset ?? 0;
+            const len = length ?? 0;
+            assert(typeof offset === "number", "offset should be number");
+            const calculatedOffset = calculateAndValidateBounds(value, offset, len);
```

Comment:
the validation is already done in `fromArrayBuffer`  so you just need to do the calculation